File: apps/backend/app/data/schemas/user_gallery.py
from datetime import datetime
from typing import Optional

from pydantic import BaseModel


class UserGalleryStageOut(BaseModel):
    stage_index: int
    variant_index: int
    title: Optional[str] = None
    description: Optional[str] = None
    video_mp4_url: Optional[str] = None
    video_webm_url: Optional[str] = None
    poster_url: Optional[str] = None
    unlocked_at: datetime


class UserGalleryScenarioOut(BaseModel):
    character_id: int
    slug: str
    name: str
    display_order: int
    poster_url: Optional[str] = None
    stages: list[UserGalleryStageOut]


class UserGalleryOut(BaseModel):
    influencer_id: str
    scenarios: list[UserGalleryScenarioOut]

File: apps/frontend/src/api/models/userGallery.ts
export interface UserGalleryStage {
  stage_index: number;
  variant_index: number;
  title: string | null;
  description: string | null;
  video_mp4_url: string | null;
  video_webm_url: string | null;
  poster_url: string | null;
  unlocked_at: string;
}

export interface UserGalleryScenario {
  character_id: number;
  slug: string;
  name: string;
  display_order: number;
  poster_url: string | null;
  stages: UserGalleryStage[];
}

export interface UserGalleryResponse {
  influencer_id: string;
  scenarios: UserGalleryScenario[];
}
